docs/structured-streaming-programming-guide.md

```diff
@@ -1647,6 +1648,26 @@ For example, sorting on the input stream is not supported, as it requires keepin
 track of all the data received in the stream. This is therefore fundamentally hard to execute 
 efficiently.
 
+### Limitation of global watermark
+
+In Append mode, some stateful operations could emit rows older than current watermark plus allowed late record delay,
+which are "late rows" in downstream stateful operations (as Spark uses global watermark) and these rows can be discarded.
+This is a limitation of global watermark, and it could bring correctness issue.
+
+Spark will check the logical plan of query and log warning when Spark detects such pattern.
+
+Any of the following stateful operation(s) after any of below stateful operations can have this issue:
```

Comment:
You refer to the below operations twice here

Reply:
Ah OK I didn't mean "following" as same as "below" but it's pretty confusing. As we mention "after" here, "following" should be removed. Thanks!